crates/runtime/src/dataconnector/listing/connector.rs

```diff
@@ -684,6 +697,65 @@ async fn get_last_modified(
     }
 }
 
+async fn verify_schema_source_path(
+    dataconnector: String,
+    dataset: &Dataset,
+    extension: &str,
+    schema_source_path: ListingTableUrl,
+    ctx: &SessionContext,
+    object_store: &Arc<dyn ObjectStore>,
+) -> DataConnectorResult<()> {
+    tracing::debug!(
+        "Verifying dataset {table_name} schema source path is valid: {schema_source_path}",
+        table_name = dataset.name
+    );
+
+    let state = ctx.state();
+    let mut file_stream = schema_source_path
+        .list_all_files(&state, object_store, "")
+        .await
+        .map_err(|err| DataConnectorError::UnableToConnectInternal {
+            dataconnector: dataconnector.clone(),
+            connector_component: ConnectorComponent::from(dataset),
+            source: err.into(),
+        })?;
+
+    let mut scanned_files = 0;
+
+    while let Some(file) =
+        file_stream
+            .try_next()
+            .await
+            .map_err(|err| DataConnectorError::UnableToConnectInternal {
+                dataconnector: dataconnector.clone(),
+                connector_component: ConnectorComponent::from(dataset),
+                source: err.into(),
+            })?
+    {
+        if let Some(ext) = file.location.extension() {
+            if format!(".{ext}") == extension {
+                return Ok(());
+            }
+        };
+
+        scanned_files += 1;
+        if scanned_files > SCHEMA_SOURCE_PATH_FILE_SCAN_LIMIT {
```

Comment:
Concern here is that if `schema_source_path` containing too many files then schema inference will take a lot of time. Should we report this as well? Try to list first N files and if num of files larger than warn